---
title: "EdgeR analysis of Arabidopsis stress RNAseq data with GSE"
author: "M. Naoumkina"
date: "`r Sys.Date()`"
output: 
  html_document: 
    toc: true
    number_sections: true
    code_download: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## EdgeR analysis of RNAseq data of Arabidopsis stress experiment

The RNAseq count data from transcription profiling experiment by high throughput sequencing of Arabidopsis plants in response to a combination of salt and heat stress have been downloaded from [Expression Atlas website](https://www.ebi.ac.uk/gxa/experiments/E-GEOD-72806/Downloads).

```{r load-edgeR, message=FALSE}
# Load edgeR
library(edgeR)
```

### Load the data and create DGE list

```{r load-data}
# Load the count data
raw_counts <- read.delim("E-GEOD-72806-raw-counts.tsv", stringsAsFactors = FALSE)
# Preserve gene ID and gene name
gene_info <- raw_counts[, c("Gene.ID", "Gene.Name")]
# Extract count matrix (assuming counts start from column 3 onward)
count_matrix <- raw_counts[, 3:ncol(raw_counts)]
# Set rownames to Gene.ID
rownames(count_matrix) <- raw_counts$Gene.ID
```

```{r}
# Load metadata
metadata <- read.delim("E-GEOD-72806-experiment-design.tsv", stringsAsFactors = FALSE)
colnames(metadata)
# Extract condition factor from metadata
condition <- factor(metadata[["Factor.Value.environmental.stress."]],
                    levels = c("none", "heat stress", "salt stress", "salt and heat stress"),
                    labels = c("control", "heat", "salt", "salt_heat"))
# Reorder columns in count matrix to match metadata
counts <- count_matrix[, metadata$Run]
# Ensure sample order matches
counts <- counts[, metadata$Run]
length(condition) == ncol(counts)  # Should return TRUE
```
```{r}
# Create DGEList object
dge <- DGEList(counts = counts, group = condition)
```

